### PR TITLE
Add Buddy support

### PR DESCRIPTION
**Pull request by akondas**
In this PR:
- [x] added support for [Buddy](https://buddy.works/)

I can create also a pipeline in Buddy and add a badge to readme (but some permissions to repo is required).

### CODE REVIEW

**Comment by Lctrs** [comment]
@akondas Could you rebase your work and add the new `isPullRequest()` method introduced in #43 please ?

**Comment by akondas** [comment]
Rebase completed  :heavy_check_mark:

File: composer.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
         "interface",
         "appveyor",
         "bamboo",
+        "buddy-works",
```
Comment (OndraM) [inline composer.json]:
```suggestion
        "buddy",
```

It is the official name, I guess.

File: tests/phpt/Ci/CiDetector_Buddy.phpt [hunk 2]
```diff
@@ -0,0 +1,95 @@
+--TEST--
+Detect properties of Buddy
```
Comment (OndraM) [inline tests/phpt/Ci/CiDetector_Buddy.phpt]:
```suggestion
Buddy: Detect properties
```

I meainwhile unified the test names.

File: tests/phpt/Ci/CiDetector_BuddyPullRequest.phpt [hunk 3]
```diff
@@ -0,0 +1,60 @@
+--TEST--
+Detect properties of Buddy
```
Comment (OndraM) [inline tests/phpt/Ci/CiDetector_BuddyPullRequest.phpt]: Dtto:

```suggestion
Buddy: Detect properties of PR build
```

**Review by OndraM — CHANGES_REQUESTED** [review]
Great, nice job @akondas :+1: . 

Please just a minor unification. Also if you could rebase the build again on master - I've just added code coverage reports to build checks.

**Comment by akondas** [comment]
So after 100% of coverage you must add mutation tests :wink:

**Review by OndraM — APPROVED** [review]

**Comment by OndraM** [comment]
Thank you, @akondas!

**Pull request merged**